identity/aziot-identityd/src/http/get_update_or_delete_module_identity.rs

```diff
@@ -36,75 +37,70 @@ impl http_common::server::Route for Route {
+		let mut inner = self.inner.lock().await;
+		let inner = &mut *inner;
+
+		let user = aziot_identityd::auth::Uid(0);
+		let auth_id = match inner.authenticator.authenticate(user) {
+			Ok(auth_id) => auth_id,
+			Err(err) => return Err(super::to_http_error(&err)),
+		};
+
+		//TODO: get uid from UDS
+		match inner.delete_identity(auth_id, "aziot", &self.module_id).await {
+			Ok(()) => (),
+			Err(err) => return Err(super::to_http_error(&err)),
+		}
+
+		Ok((hyper::StatusCode::NO_CONTENT, None))
 	}
 
 	type GetResponse = aziot_identity_common_http::get_module_identity::Response;
-	fn get(self) -> http_common::server::RouteResponse<Self::GetResponse> {
-		Box::pin(async move {
-			let mut inner = self.inner.lock().await;
-			let inner = &mut *inner;
-
-			let user = aziot_identityd::auth::Uid(0);
-			let auth_id = match inner.authenticator.authenticate(user) {
-				Ok(auth_id) => auth_id,
-				Err(err) => return Err(super::to_http_error(&err)),
-			};
-
-			//TODO: get uid from UDS
-			let identity = match inner.get_identity(auth_id, "aziot", &self.module_id).await {
-				Ok(v) => v,
-				Err(err) => return Err(super::to_http_error(&err)),
-			};
-			let res = aziot_identity_common_http::get_module_identity::Response {
-				identity,
-			};
-			Ok((hyper::StatusCode::OK, res))
-		})
+	async fn get(self) -> http_common::server::RouteResponse<Self::GetResponse> {
+		let mut inner = self.inner.lock().await;
+		let inner = &mut *inner;
+
+		let user = aziot_identityd::auth::Uid(0);
+		let auth_id = match inner.authenticator.authenticate(user) {
+			Ok(auth_id) => auth_id,
+			Err(err) => return Err(super::to_http_error(&err)),
+		};
+
+		//TODO: get uid from UDS
+		let identity = match inner.get_identity(auth_id, "aziot", &self.module_id).await {
+			Ok(v) => v,
+			Err(err) => return Err(super::to_http_error(&err)),
+		};
+		let res = aziot_identity_common_http::get_module_identity::Response {
+			identity,
+		};
+		Ok((hyper::StatusCode::OK, res))
 	}
 
 	type PostBody = serde::de::IgnoredAny;
 	type PostResponse = ();
 
 	type PutBody = serde::de::IgnoredAny;
 	type PutResponse = aziot_identity_common_http::update_module_identity::Response;
-	fn put(self, _body: Self::PutBody) -> http_common::server::RouteResponse<Self::PutResponse> {
-		Box::pin(async move {
-			let mut inner = self.inner.lock().await;
-			let inner = &mut *inner;
-
-			let user = aziot_identityd::auth::Uid(0);
-			let auth_id = match inner.authenticator.authenticate(user) {
-				Ok(auth_id) => auth_id,
-				Err(err) => return Err(super::to_http_error(&err)),
-			};
-
-			let identity = match inner.update_identity(auth_id, "aziot", &self.module_id).await {
-				Ok(v) => v,
-				Err(err) => return Err(super::to_http_error(&err)),
-			};
-			let res = aziot_identity_common_http::update_module_identity::Response {
-				identity,
-			};
-			Ok((hyper::StatusCode::OK, res))
-		})
+	#[allow(clippy::needless_pass_by_value)]
```

Comment:
As discussed, add a comment here about why this is being done. Something like this is sufficient:

```rust
// clippy fires this lint for the `_body` parameter of the inner fn in the `async-trait` expansion.
// It's not clear why clippy does this, especially since it doesn't raise it for other functions
// that also ignore their `_body` parameter `fn delete` above.
//
// So suppress it manually.
```